dptb/data/dataset/_default_dataset.py

```
@@ -72,23 +77,49 @@ def __init__(self,
+                if os.path.exists(os.path.join(self.root, "wk.npy")):
+                    # two shape cases: (nk,) or (nframe, nk)
+                    wk = np.load(os.path.join(self.root, "wk.npy"))
+                    if wk.ndim == 1:
+                        # copy it to all frames
+                        wk = np.expand_dims(wk, axis=0)
+                        self.data["wk"] = np.broadcast_to(wk, (self.info["nframes"], wk.shape[1]))
+                    elif wk.ndim == 2 and wk.shape[0] == self.info["nframes"]:
+                        # array of kpoint weights, (nframes, nkpoints)
+                        self.data["wk"] = wk
+                    else:
+                        raise ValueError("Wrong kpoint weight dimensions.")                    
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)